sarc/jobs/series.py

```diff
@@ -218,6 +220,135 @@ def compute_job_statistics(job: SlurmJob):
+        they will have in the DataFrame.
+    clip_time: bool
+        Whether the duration time of the jobs should be clipped within `start` and `end`.
+        Defaults to False.
+    callback: Callable
+        Callable taking the list of job dictionaries in the format it would be included in the DataFrame.
+    jobs_args
+        Arguments to be passed to `get_jobs` to query jobs from the database.
+    """
+
+    # If fields is a list, convert it to a renaming dict with same old and new names.
+    if isinstance(fields, list):
+        fields = {key: key for key in fields}
+
+    start = jobs_args.get("start", None)
+    end = jobs_args.get("end", None)
+
+    total = count_jobs(**jobs_args)
+
+    rows = []
+    now = datetime.now(tz=MTL)
+    # Fetch all jobs from the clusters
+    for job in tqdm(get_jobs(**jobs_args), total=total, desc="load job series"):
+        if job.end_time is None:
+            job.end_time = now
+
+        # For some reason start time is not reliable, often equal to submit time,
+        # so we infer it based on end_time and elapsed_time.
+        job.start_time = job.end_time - timedelta(seconds=job.elapsed_time)
+
+        unclipped_start = None
+        unclipped_end = None
+        if clip_time:
+            if start is None:
+                raise ValueError("Clip time: missing start")
+            if end is None:
+                raise ValueError("Clip time: missing end")
+            # Clip the job to the time range we are interested in.
+            # NOTE: This should perhaps be helper function for dataframes so that we don't force clipping raw data
+            #       during loading.
+            unclipped_start = job.start_time
+            job.start_time = max(job.start_time, start)
+            unclipped_end = job.end_time
+            job.end_time = min(job.end_time, end)
+            # Could be negative if job started after end. We don't want to filter
+            # them out because they have been submitted before end, and we want to
+            # compute their wait time.
+            job.elapsed_time = max((job.end_time - job.start_time).total_seconds(), 0)
+
+        if job.stored_statistics is None:
+            job_series = DUMMY_STATS.copy()
+        else:
+            job_series = job.stored_statistics.dict()
+            job_series = {k: select_stat(k, v) for k, v in job_series.items()}
+
+        # TODO: Why is it possible to have billing smaller than gres_gpu???
+        billing = job.allocated.billing or 0
+        gres_gpu = job.allocated.gres_gpu or 0
+        if gres_gpu:
+            job_series["gpu_allocated"] = max(billing, gres_gpu)
+            job_series["cpu"] = job.allocated.cpu
+        else:
+            job_series["gpu_allocated"] = 0
+            job_series["cpu"] = (
+                max(billing, job.allocated.cpu) if job.allocated.cpu else 0
+            )
+        job_series["gpu_requested"] = gres_gpu
```

Comment:
Looks like there is a bug here (probably coming from my initial example, sorry!). It should be `job.requested.gres_gpu`.

Reply:
Should I change `billing` too ? https://github.com/mila-iqia/SARC/blob/sarc-229/sarc/jobs/series.py#L310

```
        billing = job.allocated.billing or 0
        gres_gpu = job.allocated.gres_gpu or 0
```

Reply:
No, for billing it's the allocated values. That's good.

Reply:
OK ! Done: https://github.com/mila-iqia/SARC/pull/89/commits/6e0d8f241ef790575fe54762570c8f6c1c2d4d48

Reply:
Oups, je voulais dire que c'était correct, faut laisser comme c'était. :sweat_smile:

Reply:
Actuellement, c'est comme ceci;

```
        billing = job.allocated.billing or 0
        gres_gpu = job.requested.gres_gpu or 0
```

Que faut-il changer ?